src/models/SDKError.model.ts

```diff
@@ -0,0 +1,19 @@
+export interface SDKErrorParams {
+  message?: string;
+  code?: number;
+  data?: unknown;
+}
+export class SDKError extends Error {
+  readonly code?: number;
+  readonly data?: unknown;
+  readonly originalError: Error;
+
+  constructor(params: SDKErrorParams) {
+    super(params.message);
+
+    this.name = 'SDKError';
+    Error.captureStackTrace?.(this, SDKError);
```

Comment:
Do we need the optional chaining operator here? I also wouldn't call the arguments for this method `params`

Reply:
Removed the chaining operator, renamed the args & matching interface

Reply:
Think we need the optional chain here - this is a V8 specific method so may not be callable in other engines
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error#static_methods